components/datetime/src/neo.rs

```diff
@@ -427,12 +427,17 @@ impl<C: CldrCalendar, FSet: DateTimeNamesMarker> FixedCalendarDateTimeFormatter<
     /// [`ZonedDateTime`]: crate::input::ZonedDateTime
     /// [`YMD`]: crate::fieldsets::YMD
     /// [`format_unchecked`]: Self::format_unchecked
-    pub fn format_unchecked(&self, datetime: DateTimeInputUnchecked) -> FormattedDateTimeUnchecked {
-        FormattedDateTimeUnchecked {
+    pub fn format_unchecked(&self, datetime: DateTimeInputUnchecked) -> Result<FormattedDateTimeUnchecked, MismatchedCalendarError> {
+        if let Some(kind) = datetime.calendar_kind {
+            if !C::can_format_kind(kind) {
```

Comment:
issue: I kinda think this check should be done by the user: https://github.com/unicode-org/icu4x/issues/6486#issuecomment-2822422521